jwst/lib/signal_slot.py

```diff
@@ -1,61 +1,57 @@
-""" A signal/slot implementation
-"""
+"""A signal/slot implementation."""
+
 from collections import namedtuple
 import inspect
 import logging
 
 
-__all__ = ['Signal',
-           'Signals',
-           'SignalsNotAClass']
+__all__ = ["Signal", "Signals", "SignalsNotAClass"]
 
 # Configure logging
 logger = logging.getLogger(__name__)
 logger.addHandler(logging.NullHandler())
 
-"""Slot data structure
-"""
-Slot = namedtuple('Slot', ['func', 'single_shot'])
-
+Slot = namedtuple("Slot", ["func", "single_shot"])
+"""Slot data structure."""
 
-class Signal():
-    """Signal
 
+class Signal:
+    """
     A Signal, when triggered, call the connected slots.
 
     Parameters
     ----------
-    funcs: func[, ...]
+    *funcs : func[, ...]
         Remaining arguments will be functions to connect
         to this signal.
 
     Attributes
     ----------
-    enabled: bool
+    enabled : bool
         If True, the slots are called. Otherwise, nothing
         happens when triggered.
-
     """
 
     def __init__(self, *funcs):
-        self._slots = list()
+        self._slots = []
         self._enabled = True
-        self._states = list()
+        self._states = []
 
         for func in funcs:
             self.connect(func)
 
     def emit(self, *args, **kwargs):
-        """Invoke slots attached to the signal
+        """
+        Invoke slots attached to the signal.
 
         No return of results is expected.
 
         Parameters
         ----------
-        args: (arg[, ...])
+        *args : (arg[, ...])
```

Comment:
I think `args` should be `list` and `kwargs` should be `dict`, here and in the next few functions.

```suggestion
        *args : list
```